### ISSUE TITLE
Feature Request: Callback to Reset Mutation Status

### ISSUE THREAD

**Issue by bradharms** [reduxjs/redux-toolkit#1455]
I would like for mutation hooks to return a callback to reset mutation status back to their initial state so that the isSuccess, isError, etc. flags are all false.

The rational is for cases like modals whose contents change in response to mutation status change where the modal may need to be re-opened after closing. It's possible to get this behavior by restructuring your component hierarchy so that the hook's component gets unmounted, or by abstracting access to state, but the first isn't always possible and the second is messier than simply resetting the mutation status as a counter to firing the mutation's trigger function.

Example:

```tsx
import React from 'react';
import {useAddToCartMutation} from './api';
import Modal from './Modal';

interface Props {
  isOpen?: boolean;
  onClose?: () => void; // Consumer will use this to set isOpen to false
}

const AddToCartModal = (props: Props) => {
  const [addToCart, addToCartMutation] = useAddToCartMutation();

  const handleAddToCart = () => { addToCart(); }
  
  const handleClose = () => { setIsOpen(false); props.onClose(); }

  useEffect(() => {
   if (props.isOpen) {
     addToCartMutation.reset(); // <--- HERE
   }
  }, [props.isOpen]);

  return <Modal open={props.isOpen}>
    {addToCartMutation.isFetching ? <div>Adding to cart...</div> :
     addToCartMutation.isError ? <><div>An error occurred</div><div><button onClick={handleClose}>Close</div> :
     addToCartMutation.isSuccess ? <><div>Thank you!</div><div><button onClick={handleClose}>Close</div> :
     <button onClick={handleAddToCart}>Add to Cart</button>
  </Modal>
};
```

**Comment by sebastienlabine** [comment]
is there a workaround right now to reset the mutation state?

**Comment by phryneas** [comment]
Triggering a new mutation. Otherwise you'll just have to track that as a local state variable yourself in some way.

**Comment by danielvarcas** [comment]
I would really appreciate this feature too. The workaround I've been using is putting the mutation inside of a component which gets unmounted. However, that component is reusable in places which require different mutations, so I have to pass a prop stating which component to use, which is pretty messy.

An `isFetching` state similar to queries is exactly what I need for my use case. Being able to reset the mutation would be useful too.

**Comment by markerikson** [comment]
Available in the 1.7 betas.
